test/components/members/table_component_test.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module Members
+  class TableComponentTest < ViewComponent::TestCase
+    test 'Should render a table of members' do
+      namespace = groups(:group_one)
+      user = users(:john_doe)
+      members = Member.for_namespace_and_ancestors(namespace).not_expired
+
+      with_request_url '/-/groups/group-1/-/members' do
+        q = Member.ransack({ s: 'user_email asc', user_email_cont: 'adm' })
```

Comment:
I think this should be done in a separate test case since you are filtering the listing. This test should just display all the members and assert that the correct info is displayed

Reply:
I wasn't trying to test searching and sorting here. That has been done within the system tests already. This component renders the members that are passed to it. If I pass q.result as the members list to the component, I would just be checking that the passed list is the same list that is being rendered and I'm already doing that. Maybe I should remove the q parameters instead?